### PR TITLE
Added Orthocode keyboard

### PR DESCRIPTION
**Pull request by Jrodna**
## Description

Added config and default keymap for the Orthocode keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/orthocode/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            case 10:
+                SEND_STRING("g");
+                break;
+            case 11:
+                SEND_STRING(" ");
+                break;
+            }
+            frogchar = (frogchar + 1) % 12;
+        } else {
+
```
Comment (drashna) [inline keyboards/orthocode/keymaps/default/keymap.c:142]:
```suggestion
```

File: keyboards/orthocode/rules.mk [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+F_CPU = 16000000
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/orthocode/rules.mk:14]:
```suggestion
```

File: keyboards/orthocode/rules.mk [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (drashna) [inline keyboards/orthocode/rules.mk:35]:
```suggestion
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output 
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/orthocode/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        }
+        break;
+    }
+    return true;
+
+}
+
+
+#ifdef ENCODER_ENABLE
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (drashna) [inline keyboards/orthocode/keymaps/default/keymap.c:152]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/orthocode/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    // Volume control
+    if (clockwise) {
+        tap_code(KC_VOLD);
+    } else {
+        tap_code(KC_VOLU);
+    }
+}
```
Comment (drashna) [inline keyboards/orthocode/keymaps/default/keymap.c:159]:
```suggestion
    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: .vscode/settings.json [hunk 6]
```diff
@@ -17,7 +17,9 @@
         "xstddef": "c",
         "type_traits": "c",
         "utility": "c",
-        "ranges": "c"
+        "ranges": "c",
```
Comment (drashna) [inline .vscode/settings.json:20]: Please revert this change

File: keyboards/orthocode/rules.mk [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/orthocode/rules.mk:26]:
```suggestion
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/orthocode/rules.mk [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no        # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (fauxpark) [inline keyboards/orthocode/rules.mk:19]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```

File: keyboards/orthocode/rules.mk [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output 
+ENCODER_ENABLE = yes       # Enables the use of one or more encoders
+
+
```
Comment (fauxpark) [inline keyboards/orthocode/rules.mk:29]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 6 above)
Comment (Jrodna) [reply .vscode/settings.json]: have reverted these on branch

File: keyboards/orthocode/rules.mk [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = yes        # Enables the use of one or more encoders
+
+LAYOUTS = orthocode
```
Comment (fauxpark) [inline keyboards/orthocode/rules.mk:28]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/orthocode/rules.mk (hunk 9 above)
Comment (Jrodna) [reply keyboards/orthocode/rules.mk]: I have made the requested changes

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!
